Source/SpatialGDK/Private/SpatialInteropPipelineBlock.cpp

```diff
@@ -187,6 +187,21 @@ void USpatialInteropPipelineBlock::LeaveCriticalSection()
 		RemoveEntityImpl(PendingRemoveEntity);
 	}
 
+	// Notify authority changes
+	for (auto& PendingAuthorityChange : PendingAuthorityChanges)
```

Comment:
Move this into AuthorityChangeImpl and call it from ChangeAuthority() when we're not in a critical section.

Reply:
Good point, will do.